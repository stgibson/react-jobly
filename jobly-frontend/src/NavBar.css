.NavBar {
  margin-bottom: 10px;
  padding: 15px;
  display: flex;
  background-color: white;
}

.NavBar a {
  color: blue;
  text-decoration: none; /* Learned how to remove underline for links at https://www.w3schools.com/css/css_link.asp */
}

.NavBar a:hover {
  text-decoration: none;
}

.NavBar a.active {
  color: black;
}

.NavBar-brand {
  margin-right: auto;
  margin-left: 10px;
}

.NavBar-link {
  margin-right: 10px;
}