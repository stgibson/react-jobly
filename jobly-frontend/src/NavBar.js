import React from "react";
import { Link, NavLink } from "react-router-dom";
import "./NavBar.css";

/**
 * Component for displaying navbar
 * @param {Object{Object{string|boolean|Object{number}}}} param0
 * @returns JSX code for rendering navbar
 */
const NavBar = ({ user }) => {
  if (user) {
    return (
      <div className="NavBar">
        <Link to="/">Jobly</Link>
        <NavLink to="/companies">Companies</NavLink>
        <NavLink to="/jobs">Jobs</NavLink>
        <NavLink to="/profile">Profile</NavLink>
      </div>
    );
  }
  return (
    <div className="NavBar">
      <Link className="NavBar-brand" to="/">Jobly</Link>
      <NavLink className="NavBar-link" to="/login">Login</NavLink>
      <NavLink className="NavBar-link" to="/signup">Sign Up</NavLink>
    </div>
  );
};

export default NavBar;