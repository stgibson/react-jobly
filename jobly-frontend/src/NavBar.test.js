import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import NavBar from "./NavBar";
import CurrentUserContext from "./CurrentUserContext";

const currentUser = {
  username: "test",
  firstName: "Test",
  lastName: "User",
  email: "test@gmail.com",
  isAdmin: false,
  jobs: {}
}

it("renders without crashing", () => {
  render(<MemoryRouter><NavBar /></MemoryRouter>);
});

it("matches snapshot", () => {
  const { asFragment } = render(<MemoryRouter><NavBar /></MemoryRouter>);
  
  expect(asFragment()).toMatchSnapshot();
});

it("only shows Login & Sign Up when no user is passed", () => {
  const { queryByText } =
    render(<MemoryRouter><NavBar /></MemoryRouter>);
  
  expect(queryByText("Login")).toBeInTheDocument();
  expect(queryByText("Sign Up")).toBeInTheDocument();
  expect(queryByText("Companies")).not.toBeInTheDocument();
  expect(queryByText("Jobs")).not.toBeInTheDocument();
  expect(queryByText("Profile")).not.toBeInTheDocument();
});

it("only shows Companies, Jobs, & Profile when user is passed", () => {
  const { queryByText } =
    render(
      <CurrentUserContext.Provider value={ currentUser }>
        <MemoryRouter><NavBar /></MemoryRouter>
      </CurrentUserContext.Provider>
    );
  
  expect(queryByText("Companies")).toBeInTheDocument();
  expect(queryByText("Jobs")).toBeInTheDocument();
  expect(queryByText("Profile")).toBeInTheDocument();
  expect(queryByText("Login")).not.toBeInTheDocument();
  expect(queryByText("Sign Up")).not.toBeInTheDocument();
});