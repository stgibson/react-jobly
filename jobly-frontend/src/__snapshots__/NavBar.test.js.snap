// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="NavBar"
  >
    <a
      class="NavBar-brand"
      href="/"
    >
      Jobly
    </a>
    <a
      class="NavBar-link"
      href="/login"
    >
      Login
    </a>
    <a
      class="NavBar-link"
      href="/signup"
    >
      Sign Up
    </a>
  </div>
</DocumentFragment>
`;
